antarest/storage/service.py

```diff
@@ -1,328 +1,184 @@
+        self._check_user_permission(params.user, uuid)
+        self.importer_service.upload_matrix(path, data)
+
+    def import_study(
+        self, stream: IO[bytes], params: StorageServiceParameters
+    ) -> str:
+        uuid = self.importer_service.import_study(stream)
+        self._save_metadata(uuid, params.user)
         return uuid
 
-    def import_output(self, uuid: str, stream: IO[bytes]) -> JSON:
-        path_output = (
-            Path(self.path_to_studies) / uuid / "output" / "imported_output"
+    def import_output(
+        self, uuid: str, stream: IO[bytes], params: StorageServiceParameters
+    ) -> JSON:
+        self._check_user_permission(params.user, uuid)
+        res = self.importer_service.import_output(uuid, stream)
+        return res
+
+    def edit_study(
+        self, route: str, new: JSON, params: StorageServiceParameters
+    ) -> JSON:
+        uuid, _, _ = self.study_service.extract_info_from_url(route)
+        self._check_user_permission(params.user, uuid)
+        return self.study_service.edit_study(route, new)
+
+    def _save_metadata(self, uuid: str, user: Optional[User]) -> None:
+        if not user:
+            raise UserHasNotPermissionError
+
+        info = self.study_service.get_study_information(uuid)["antares"]
+        meta = Metadata(
+            id=uuid,
+            name=info["caption"],
+            version=info["version"],
+            author=info["author"],
+            created_at=datetime.fromtimestamp(info["created"]),
+            updated_at=datetime.fromtimestamp(info["lastsave"]),
+            users=[user],
         )
-        path_output.mkdir()
-        StorageService.extract_zip(stream, path_output)
-
-        ini_reader = IniReader()
-        info_antares_output = ini_reader.read(
-            path_output / "info.antares-output"
-        )["general"]
+        self.repository.save(meta)
 
-        date = datetime.fromtimestamp(
-            int(info_antares_output["timestamp"])
-        ).strftime("%Y%m%d-%H%M")
+    def _check_user_permission(
+        self, user: Optional[User], uuid: str, raising: bool = True
+    ) -> bool:
+        def check(user: Optional[User], uuid: str) -> None:
+            if not user:
+                raise UserHasNotPermissionError()
 
-        mode = "eco" if info_antares_output["mode"] == "Economy" else "adq"
-        name = (
-            f"-{info_antares_output['name']}"
-            if info_antares_output["name"]
-            else ""
-        )
+            if user.role == Role.ADMIN:
+                return
 
-        output_name = f"{date}{mode}{name}"
-        path_output.rename(Path(path_output.parent, output_name))
+            md = self.repository.get(uuid)
+            if not md:
+                logging.warning(f"Study {uuid} not found in metadata db")
+                # TODO don't raise error. Study can be added manually by user
+                raise StudyNotFoundError(uuid)
```

Comment:
so... it should return (to skip the user check) and not throw an exception.
Also the logging should be defined in the beginning of the file with logger = logging.get_logger(__name__)

Reply:
ok for logging. I prefere to raise an error now. When we have a clear view of files watcher and authorization logic, I will suppress it.